File: Ch03. Basic Data Structures/Java code/StackAndQueueAsLinkedList/Stack.java

public class Stack {

	private Node stackPointer;
	
	public Stack() {
		this.stackPointer = null;
	}
	public boolean isEmpty() {
		return this.stackPointer == null;
	}
	public int size() {
		int counter = 0;
		for(Node tmp = this.stackPointer; tmp != null; tmp = tmp.next)
			counter++;
		return counter;
	}
	public void push(int i) {
		this.stackPointer = new Node(i, this.stackPointer);
	}
	public Integer pop() {
		if(this.isEmpty())
			return null;
		else {
			Integer i = this.stackPointer.info;
			this.stackPointer = this.stackPointer.next;
			return i;
		}
	}
	public Integer peek() {
		if(this.isEmpty())
			return null;
		else {
			Integer i = this.stackPointer.info;
			return i;
		}
	}
}

File: Ch03. Basic Data Structures/Stack/stack.py
class Stack:
    def __init__(self):
        self.stack: list = []

    def isEmpty(self)->bool:
        return len(self.stack) == 0

    def size(self)->int:
        return len(self.stack)

    def push(self, item):
        self.stack.append(item)

    def pop(self):
        return self.stack.pop()

    def peek(self):
        return self.stack[len(self.stack) - 1]

